fineract-loan/src/main/java/org/apache/fineract/portfolio/interestpauses/service/InterestPauseReadPlatformServiceImpl.java

```diff
@@ -0,0 +1,151 @@
+import org.apache.fineract.infrastructure.core.domain.ExternalId;
+import org.apache.fineract.infrastructure.core.exception.GeneralPlatformDomainRuleException;
+import org.apache.fineract.infrastructure.core.service.DateUtils;
+import org.apache.fineract.infrastructure.security.service.PlatformSecurityContext;
+import org.apache.fineract.portfolio.interestpauses.data.InterestPauseResponseDto;
+import org.apache.fineract.portfolio.loanaccount.data.LoanTermVariationsData;
+import org.apache.fineract.portfolio.loanaccount.domain.Loan;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanRepository;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanTermVariationType;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanTermVariations;
+import org.apache.fineract.portfolio.loanaccount.exception.LoanNotFoundException;
+import org.apache.fineract.portfolio.loanaccount.rescheduleloan.domain.LoanTermVariationsRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class InterestPauseReadPlatformServiceImpl implements InterestPauseReadPlatformService {
+
+    private final PlatformSecurityContext context;
+    private final LoanTermVariationsRepository loanTermVariationsRepository;
+    private final LoanRepository loanRepository;
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<InterestPauseResponseDto> retrieveInterestPauses(Long loanId) {
+        this.context.authenticatedUser();
+
+        List<LoanTermVariationsData> variations = this.loanTermVariationsRepository.findLoanTermVariationsByLoanIdAndTermType(loanId,
+                LoanTermVariationType.INTEREST_PAUSE.getValue());
+
+        return mapToInterestPauseResponse(variations);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<InterestPauseResponseDto> retrieveInterestPausesByExternalId(String loanExternalId) {
+        this.context.authenticatedUser();
+
+        List<LoanTermVariationsData> variations = this.loanTermVariationsRepository.findLoanTermVariationsByExternalLoanIdAndTermType(
+                new ExternalId(loanExternalId), LoanTermVariationType.INTEREST_PAUSE.getValue());
+
+        return mapToInterestPauseResponse(variations);
+    }
+
+    @Override
+    @Transactional
+    public CommandProcessingResult createInterestPause(ExternalId loanExternalId, LocalDate startDate, LocalDate endDate, String dateFormat,
+            String locale) {
+        this.context.authenticatedUser();
+
+        Long loanId = loanRepository.findIdByExternalId(loanExternalId);
+        if (loanId == null) {
+            throw new LoanNotFoundException("Loan not found with External ID: " + loanExternalId);
+        }
+
+        Loan loan = loanRepository.findByExternalId(loanExternalId).orElseThrow(() -> new LoanNotFoundException(loanExternalId));
```

Comment:
Same as above.. and not needed do check twice the things!

Reply:
Fixed